lib/src/client.dart

```diff
@@ -2315,6 +2316,10 @@ class Client extends MatrixApi {
       }
       onEvent.add(update);
 
+      if (event.type == EventTypes.SpaceChild) {
```

Comment:
You also need to trigger this, when a parent is changed, since you don't need both relations.